luma/core/device.py

```diff
@@ -188,35 +189,38 @@ class linux_framebuffer(device):
     leverage kernel-based display drivers.
 
     .. note:
-        Currently only supports 16-bit and 24-bit RGB color depths.
+        Currently only supports 16-bit RGB, 24-bit RGB/BGR and 32-bit RGBA/BGRA color depths.
 
     :param device: the Linux framebuffer device (e.g. `/dev/fb0`). If no device
         is given, the device is determined from the `FRAMEBUFFER` environmental
         variable instead. See https://www.kernel.org/doc/html/latest/fb/framebuffer.html
         for more details.
     :param framebuffer: Framebuffer rendering strategy, currently instances of
         ``diff_to_previous`` (default, if not specified) or ``full_frame``.
+    :param bgr: Set to ``True`` if device pixels are BGR order (rather than RGB). Note:
+        this flag is currently supported on 24 and 32-bit color depth devices only.
 
     .. versionadded:: 2.0.0
     """
 
-    def __init__(self, device=None, framebuffer=None, **kwargs):
+    def __init__(self, device=None, framebuffer=None, bgr=False, **kwargs):
         super(linux_framebuffer, self).__init__(serial_interface=noop())
         self.id = self.__get_display_id(device)
         (width, height) = self.__config("virtual_size")
         self.bits_per_pixel = next(self.__config("bits_per_pixel"))
         image_converters = {
-            16: self.__toRGB565,
-            24: self.__toRGB,
+            (16, False): self.__toRGB565,
+            (24, False): self.__toRGB,
+            (24, True): self.__toBGR,
+            (32, False): self.__toRGBA,
+            (32, True): self.__toBGRA,
         }
-        assert self.bits_per_pixel in image_converters, f"Unsupported bit-depth: {self.bits_per_pixel}"
-        self.__image_converter = image_converters[self.bits_per_pixel]
+        assert (self.bits_per_pixel, bgr) in image_converters, f"Unsupported bit-depth: {self.bits_per_pixel}"
+        self.__image_converter = image_converters[(self.bits_per_pixel, bgr)]
 
         self.framebuffer = framebuffer or diff_to_previous(num_segments=16)
         self.capabilities(width, height, rotate=0, mode="RGB")
-
-        path = f"/dev/fb{self.id}"
-        self.__file_handle = open(path, "wb")
+        self.__file_handle = open(f"/dev/fb{self.id}", "wb")
```

Comment:
an explicit comment somewhere that this handle doesn't close until `cleanup`

Reply:
fixed